HARJA-1585 - Päivitä Reagent 0.10.0 -> 1.1.1

K1ll3rF0x opened the pull request:
**TODO**: 
* Kannattaa vielä manuaalitestailla laajemmin näkymiä ja pohtia saisiko selaimen lokille tulostuvan komponentinrenderöintivirheen korjattua ennen mergeä. 

* Korjattava havainto 23-10-2025, kuvaus alla.

----


Tarkemmassa testauksessa havaittuja virheitä:
* [ :heavy_check_mark: Korjattu ] **:numero-pohjaiset input-kentät eivät toimi kunnolla**
   * Kaikki muut input kentät toimivat normaalisti, mutta numerokenttien kanssa on ongelmia
   * Merkkien pyyhkiminen kentässä johtaa **maximum call stack size** exceeded virheeseen, dataa voi kuitenkin lisätä ja numerokentän arvon voi korvata copy-pastella.
   * Virhe näyttää tapahtuvan data-atomin deref vaiheessa numero-kentän koodissa. Tämä on kummallista, koska logiikka datan muokkaamiselle on samantyylinen kuin kaikissa muissa kentissä. Jotain eroa tässä kuitenkin täytyy olla muihin kenttiin, koska vain tämä on lakannut toimista uudemmalla reagentilla.
* [ :heavy_check_mark: Korjattu ] **Välilehden valinta aiheuttaa joskus **maximum call stack size** virheen**, joka ei näytä aiheuttavan visuaalisia ongelmia
  *  Virhe taitaa johtua samantyylisestä ongelmasta kuin input-kentälläkin. Tällä on jotain tekemistä siihen miten reagentin reactioneita käytetään Harjassa ja miten atomeita käsitellään.
  
  
**23-10-2025:**

:warning: Korjattava
> Jotakin on vielä mennyt rikki näkymien tilaan liittyvien korjauksien jälkeen.
> Cypress testejä on kosahtanut tietyissä näkymissä aika paljon:
> https://github.com/finnishtransportagency/harja/runs/52447781842
> 
> Lähinnä näyttää liittyvän päällystyksiin ja paikkauksiin, ja POT2-lomakkeeseen.
> Ehkä siellä on käytetty jotain eksoottisempaa tilanhallintaa ja reaktioita.

----
Cypress-testit menevät läpi.
Vanhat phantomjs-testit valitettavasti käyttävät vanhempaa Reagenttia ja Reactia yhteensopivuusongelmien takia, joten phantom-testien läpimeno ei kerro onko uudempi Reagent toimiva.

Tehty manuaalista testausta, eikä käyttöliittymää radikaalisti rikkovia ongelmia ole löytynyt.
Selaimen lokiin tulee ainakien Harjan välilehtien vaihdosta joissakin yhteyksissä komponenttien rendaukseen liittyviä virheitä, jotka eivät näytä vaikuttavat visuaalisesti.
Ongelma voi liittyä reagentin atomien käyttöön ja wrappaukseen. Jotain on näiden suhteen todennäköisesti muuttunut uudemmassa reagentissa, ja se olisi syytä tutkia ja korjata virheet pois.

Yleisesti, näyttää siltä että Reagent 1.1.1 toimii aika hyvin yhteen Harjan nykytoteutuksen kanssa. 
Mutta, **manuaalisesta testausta on syytä tehdä vielä laajemmin ennen mergeä.**

----

**Reagent v1.0.0 muutokset:**
https://github.com/reagent-project/reagent/blob/master/CHANGELOG.md#100-2020-12-21

**Reagent v1.1.0 muutokset:**
https://github.com/reagent-project/reagent/blob/master/CHANGELOG.md#110-2021-06-05

**Reagent v1.1.1 muutokset:**
https://github.com/reagent-project/reagent/blob/master/CHANGELOG.md#111-2022-03-09

**React 17 muutokset:**
https://legacy.reactjs.org/blog/2020/10/20/react-v17.html
https://github.com/facebook/react/blob/main/CHANGELOG.md#1701-october-22-2020
https://github.com/facebook/react/blob/main/CHANGELOG.md#1702-march-22-2021

K1ll3rF0x committed:
Päivitä Reagent 0.10.0 -> 1.0.0

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

K1ll3rF0x committed:
Älä käytä deprekoituja cy.route kutsuja

K1ll3rF0x committed:
Korjaa React 17 Event Delegation muutosten aiheuttamat virheet
* https://legacy.reactjs.org/blog/2020/08/10/react-v17-rc.html#fixing-potential-issues

K1ll3rF0x committed:
Korjaa cy.intercept kutsut

K1ll3rF0x committed:
Päivitä Reagent v1.0.0 -> v1.1.1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Kokeile päivitettyä cypress-imagea

K1ll3rF0x committed:
Revert "Kokeile päivitettyä cypress-imagea"

This reverts commit cf648eb80cf22fd1e16ee0b764467f0b24042a91.

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x commented:
Tarkemmassa testauksessa havaittuja virheitä:
* **:warning: :numero-pohjaiset input-kentät eivät toimi**
   * Kaikki muut input kentät toimivat normaalisti, mutta numerokenttien kanssa on ongelmia
   * Merkkien pyyhkiminen kentässä johtaa **maximum call stack size** exceeded virheeseen
   * Virhe näyttää tapahtuvan data-atomin deref vaiheessa numero-kentän koodissa. Tämä on kummallista, koska logiikka datan muokkaamiselle on samantyylinen kuin kaikissa muissa kentissä. Jotain eroa tässä kuitenkin täytyy olla muihin kenttiin, koska vain tämä on lakannut toimista uudemmalla reagentilla.
* **Välilehden valinta aiheuttaa joskus **maximum call stack size** virheen**, joka ei näytä aiheuttavan visuaalisia ongelmia
  *  Virhe taitaa johtua samantyylisestä ongelmasta kuin input-kentälläkin. Tällä on jotain tekemistä siihen miten reagentin reactioneita käytetään Harjassa ja miten atomeita käsitellään.

rckopteri reviewed (changes requested):
Päivitän tämän request changes - malliin, koska tosiaan jotain joudutaan vielä tälle tekemään.

rckopteri committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

rckopteri committed:
Poista wrappi atomista ja korvaa se ihan tavallisella atomilla. Wrapattyä uutta atomin asetusta ei tarvita

rckopteri committed:
Poista navigaatiosta turha valilehtiatomi. Käytä reittien alkuperäistä.

rckopteri committed:
Merge branch 'upgrade-reagent-v1' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

K1ll3rF0x committed:
Korjaa :numero-input uudelle Reagent-versiolle
* :numero kentän atomien arvon asettaminen ja deref aiheutti ongelmia
uudemmassa Reagent-versiossa. Numero-kentän formatoinnin logiikka
teksti- ja data-atomien suhteen aiheutti ikuisen loopin tietyissä
tilanteissa, koskien erityisesti desimaalilukuja. Tämä ilmeni vain
uudemmassa Reagent versiossa, jossa on ilmeisesti tehty muutoksia
render-sykliin.

K1ll3rF0x committed:
Korjaa bugi :numero-kentän miinusmerkin formatoinnin suhteen
* Harjassa formatoidaan negatiivinen :numero-kenttä niin, että
miinusmerkki muuttuu taustalla matemaattiseksi miinusmerkiksi, joka on
eri asia kuin normaali hyphen "-". Tämä pitää ottaa huomioon luvun
parsinnassa ja validoinnissa.

K1ll3rF0x committed:
Älä päivitä dataa, jos negatiivinen luku on kesken (vain miinumerkki
syötetty)

K1ll3rF0x committed:
Korjaa front-end testi

K1ll3rF0x commented:
Jotakin on vielä mennyt rikki näkymien tilaan liittyvissä korjauksissa.

Cypress testejä on kosahtanut tietyissä näkymissä aika paljon:
https://github.com/finnishtransportagency/harja/runs/52447781842

Lähinnä näyttää liittyvän päällystyksiin ja paikkauksiin, ja POT2-lomakkeeseen.
Ehkä siellä on käytetty jotain eksoottisempaa tilanhallintaa ja reaktioita.

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

rckopteri committed:
Kasvata timeouttia

rckopteri committed:
Poista turhien atomien käyttöä ja refaktoroi koodia

rckopteri committed:
Poista turha logitus

K1ll3rF0x commented:
Kaikki virheet testeissä on korjattu.
Vielä viimeinen manuaalinen testikierros ja PR olisi valmis tuotantoon.

@finnishtransportagency/harja-solita

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

solita-staskila commented:
Tätä tulisi testata tällä haaralla 
https://github.com/finnishtransportagency/harja/pull/4210

K1ll3rF0x committed:
Merge branch 'develop' of github.com:finnishtransportagency/harja into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1

K1ll3rF0x committed:
Merge branch 'develop' into upgrade-reagent-v1
